src/components/Voting/Referendum/index.js

```diff
@@ -51,33 +59,52 @@ function Referendum() {
   const { motionIds } = useMotionContext();
 
   useEffect(() => {
-    if (proposalRef.current || dispatchRef.current) {
-      dispatch(identityActions.getIdentityMotions.call(Array.from(new Set(motionIds))));
-    }
+    dispatch(identityActions.getIdentityMotions.call(Array.from(new Set(motionIds))));
   }, [motionIds, dispatch, proposalRef, dispatchRef]);
 
   return (
-    <div className={cx(stylesPage.contentWrapper, styles.wrapper)}>
-      <div>
-        <h3 className={styles.title}>Referendums</h3>
+    <Collapse
+      defaultActiveKey={['referendums', 'proposals', 'dispatches']}
+      items={[
         {
-            delegatingTo
-              ? (
-                <div className={styles.proposeReferendumLine}>
-                  (
-                  Delegating to:
-                  {' '}
-                  {delegatingTo}
-                  <Button small primary onClick={handleModalOpenUndelegate}>Undelegate</Button>
-                  )
-                </div>
-              )
-              : null
-        }
-        <div className={styles.overViewCard}>
-          {democracy.democracy?.crossReferencedReferendumsData?.length > 0
-            ? democracy.democracy?.crossReferencedReferendumsData.map((referendum) => (
-              <Card className={stylesPage.overviewWrapper} key={referendum.index}>
+          key: 'referendums',
+          label: 'Referendums',
+          extra: (
+            <Flex wrap gap="15px">
+              {delegatingTo && (
+                <>
+                  <UndelegateModal
+                    delegatee={delegatingTo}
+                    onSubmitUndelegate={handleSubmitUndelegate}
+                  />
+                  <div>
+                    Delegating to:
+                    {' '}
+                    {delegatingTo}
+                  </div>
+                </>
+              )}
+              {user ? (
+                <Button
+                  onClick={() => history.pushState(router.voting.addLegislation)}
+                  primary
+                >
+                  Propose
```

Comment:
button doesnt work, it just closes/opens the referendum menu, it should go to propose legislation menu

Reply:
Replaced with navlink. Still closes menu for some unimaginable reason, but I'll fix that later